frontend/src/app/window.service.ts

```diff
@@ -1,8 +1,20 @@
+import { isPlatformBrowser } from '@angular/common';
 import { InjectionToken } from '@angular/core';
 
+interface FakeWindow {
+  ga: () => any,
+  location: Object,
+}
 
 export const WindowService = new InjectionToken<Window>('Window');
 
-export function WindowFactory(): Window {
-  return window;
+export function WindowFactory(platformId: any): Window | FakeWindow {
+  if (isPlatformBrowser(platformId)) {
+    return window;
+  } else {
+    return  {
+      ga: () => null,
```

Comment:
I guess this would solve the problem of rendering but we lose the Google Analytics script in the process, so I guess the tracking would not be enabled until frontend app starts?

Reply:
@CorneeSean - it's just for server side rendering - When already rendered html lands in user browser it will download Analitycs script into browser and standard Angular initialization will occur. 

So I believe that's better, that we mock it on server side, as it could make traffic be double counted.